api/v1alpha4/condition_consts.go

```diff
@@ -35,6 +35,9 @@ const (
 
 	// DeletedReason (Severity=Info) documents an condition not in Status=True because the underlying object was deleted.
 	DeletedReason = "Deleted"
+
+	// IncorrectExternalRefReason (Severity=Error) documents a CAPI object with an incorrect external object reference.
+	IncorrectExternalRefReason = "IncorrectExternalRef"
```

Comment:
Is this the first time we're using this kind of condition? cc @fabriziopandini 

How do we propagate this change elsewhere as well?

Reply:
AFAIK this is the first time we are exposing this reason in a condition. It seems like this failure case could occur in other controllers as well, and we might need to add the logic where it makes sense.